pkg/storage/common/common.go

```diff
@@ -83,7 +85,7 @@ func ValidateManifest(imgStore storageTypes.ImageStore, repo, reference, mediaTy
 		if err := ValidateManifestSchema(body); err != nil {
 			log.Error().Err(err).Msg("OCIv1 image manifest schema validation failed")
 
-			return "", zerr.ErrBadManifest
+			return "", zerr.ErrBadManifest.AddDetail("jsonSchemaValidation", err.Error())
```

Comment:
OK, but if we do this, don't we modify the global variable `zerr.ErrBadManifest`? I think we would need a new instance with the same error string, and we should also be able to compare that new error instance with `zerr.ErrBadManifest` in order to see if they have the same return value for .Error(). Maybe overwrite the `Is()` function for the comparison.

A separate note: We should have a similar return at line 127 in case of index validation failures. Also line 791 would need an update.